### DISCUSSION TITLE
"st.cache is deprecated." msg on "species stats"

### DISCUSSION THREAD

**Discussion by calyptura** [mcguirepr89/BirdNET-Pi#753] [Q&A] [answered]
Dear BirdNETers,

When I access the "species stats" page I'm getting some "st.cache is deprecated." messages (picture below). This is normal? How can I fix it?

Thanks!

Luciano
<img width="1437" alt="Captura de Tela 2023-02-22 às 09 02 28" src="https://user-images.githubusercontent.com/126075135/220615593-8c9c7eb4-303c-4c79-8f38-12023656020c.png">

**Comment by oester** [comment, marked as answer]
I just came across this and fixed it by doing this:
 edit scripts/plotly_streamlit.py
 change first occurrence @st.cache(allow_output_mutation=True) to @st.cache_resource   <--- no arguments
 change second occurrence @st.cache to @st.cache_data
 change third  occurrence @st.cache to @st.cache_data

Save file and the next time you go to the page it should be fixed. (no need to restart)

**Reply by calyptura** [reply]
Thanks Oester! Where do I go to edit scripts?

**Reply by oester** [reply]
In your installation directory, you'll find it in the scripts directory.

**Reply by calyptura** [reply]
Found it! Two messages disappeared but I still get one wrong message... 
Did I edit the first one properly?
Thanks!
![Captura de Tela 2023-02-22 às 19 27 21](https://user-images.githubusercontent.com/126075135/220774857-03304647-6516-44ab-9f64-8f5ea2eae053.png)
![Captura de Tela 2023-02-22 às 19 29 12](https://user-images.githubusercontent.com/126075135/220774887-1c4a5160-35a9-4de2-9472-6c40bc0d5485.png)

**Reply by oester** [reply]
Remove or comment out ("#")the first line and remove the comment on the second line so the "@" is in the first column. 

Oh, and remove the "<---- no arguments " at the end. Sorry that was a comment.

**Reply by calyptura** [reply]
It worked! Thank you very much!
